backend/lab-service/src/main/java/org/mltooling/lab/api/LabAdminApiHandler.java

```diff
@@ -237,9 +243,12 @@ public StatusMessageFormat checkWorkspace(String user) {
       }
 
       response.setSuccessfulStatus();
-      response.addMetadata(
-          "needsUpdate",
-          !CoreService.WORKSPACE.getImage().equalsIgnoreCase(workspaceService.getDockerImage()));
+      response.setData(workspaceService);
```

Comment:
This will return a lot of information about the Service (for example all its environment variables).
The admin/workspace/check endpoint actually requires NO authentication.
So an malicious unauthenticated users can get the environment variables of the admin workspace which includes the admin JWT token and therefore get full admin privileges.
Suggestions:
- Create a separate class for the return value of this function instead of reusing the "LabService" class that contains too much sensitive data
- Add authorization to the admin/workspace/check endpoint so only admins can query it for any workspace and users can only query it for their own workspace

Reply:
With regards to the suggestion of adding authorization to the endpoint, just keep in mind that it is also called from nginx and probably a change such as explicitly adding the user token to the request must be added.

Reply:
Thanks for mentioning that. We already thought that there was probably a reason to not do authentication for the endpoint.
Alternatively, we could maybe leave the check workspace as it is (without authentication and only returning true/false) and add a new endpoint /workspace/info that returns workspace information and secure it properly.

Reply:
yes, that could be an alternative. In case you want to try out the securing option, the line in nginx where the endpoint is called from Lua script is [here](https://github.com/SAP/machine-learning-lab/blob/9f1733776be8fc9ed894ce9b01bbfb382a72d0de/backend/lab-service/docker-res/nginx.conf#L247). Since you can access nginx variables there, I think you could access the lab-access token cookie / Auth Header there and set it in the lua request.

Reply:
Thanks for the heads-up and pointing us in the right direction @raethlein ! 
I added the token to the request [here](https://github.com/ericksantillanp/machine-learning-lab/blob/a8fcc3e74b2022efcc49da8868bbbabfd6b5ee1a/backend/lab-service/docker-res/nginx.conf#L249)

We also found out that when we use the authentication header with the token it shouldn't be passed as `Authorization: Bearer <TOKEN>` but rather `Authorization: <TOKEN>` . This is a little bit different than what is specified in the standard [RFC 6750](https://tools.ietf.org/html/rfc6750) for OAuth2. Is there a reason for this?

Reply:
I don't think there is an intentional reason behind this and should probably better follow the RFC standard. Off the top of my head I don't know whether changing this will break anything or whether it must be changed at multiple places, so perhaps the safest solution would be to allow the "Bearer" keyword but not force it - or force it and then keep in mind that, if something breaks, this is the reason and must be added elsewhere.